nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/listener/CommentNotificationListener.java

```diff
@@ -37,12 +41,23 @@ public void handleNotifications(Event event) {
         EventContext ctx = event.getContext();
         DocumentEventContext docCtx = (DocumentEventContext) ctx;
         if (docCtx.getSourceDocument().getType().equals("Post")
+                || docCtx.getSourceDocument().getType().equals(AnnotationConstants.ANNOTATION_DOC_TYPE)
                 || docCtx.getSourceDocument().getType().equals(CommentsConstants.COMMENT_DOC_TYPE)) {
             CommentManager commentManager = Framework.getService(CommentManager.class);
-            DocumentModel thread = commentManager.getThreadForComment(docCtx.getSourceDocument());
-            if (thread != null) {
-                Object[] args = { thread, null };
-                docCtx.setArgs(args);
+            try {
+                DocumentModel thread = commentManager.getThreadForComment(docCtx.getSourceDocument());
+                if (thread != null) {
+                    Object[] args = { thread, null };
+                    docCtx.setArgs(args);
+                }
+            } catch (IllegalArgumentException e) {
+                // CommentManagerImpl#getTopLevelCommentAncestor has been called and returned a null reference
+                if (e.getMessage().equals(CoreSession.NULL_DOC_REF) && commentManager instanceof CommentManagerImpl) {
+                    Object[] args = { docCtx.getProperty(CommentConstants.TOP_LEVEL_DOCUMENT) };
+                    docCtx.setArgs(args);
+                } else {
+                    throw e;
+                }
```

Comment:
i am wondering what if we use directly `docCtx.getProperty(CommentConstants.TOP_LEVEL_DOCUMENT)` on `docCtx.setArgs(args);` instead of the call to  `commentManager.getThreadForComment(docCtx.getSourceDocument())` if i remember from our investigation :) this listener is only added to fix the problem that in notification listener it need the `Top level document` but our `document source` of the doc context is the comment itself and it failed in the case of reply . Perhaps it will be better to rework this listeners and replace the call to `commentManager.getThreadForComment(docCtx.getSourceDocument(...` by `ocCtx.getProperty(CommentConstants.TOP_LEVEL_DOCUMENT)` and add javadoc/comment to explain it

Reply:
I'm in favor of using directly the call to docCtx.getProperty(CommentConstants.TOP_LEVEL_DOCUMENT) as you say. 
It's the document we aim at and it's already there. No need for extra calls and recursions etc...

Reply:
exactly just add some javadoc/comment on why we rely on it ... from my point of view this listener should exist its break immutability and its hard to understand that we have a listener which role is to modify what we dispatch